flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSource.java

```diff
@@ -393,7 +396,27 @@ public int hashCode() {
                 kafkaSourceBuilder.setStartingOffsets(OffsetsInitializer.latest());
                 break;
             case GROUP_OFFSETS:
-                kafkaSourceBuilder.setStartingOffsets(OffsetsInitializer.committedOffsets());
+                String offsetResetConfig =
+                        properties.getProperty(
+                                ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+                                OffsetResetStrategy.NONE.name());
+                OffsetResetStrategy offsetResetStrategy =
+                        Arrays.stream(OffsetResetStrategy.values())
+                                .filter(
+                                        ors ->
+                                                ors.name()
+                                                        .equals(
+                                                                offsetResetConfig.toUpperCase(
+                                                                        Locale.ROOT)))
```

Comment:
Nit: This is a bit hard to read maybe extract a method.

Reply:
fixed